anthos-bm-gcp-terraform/main.tf

```diff
@@ -0,0 +1,188 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "3.58.0"
+    }
+    google-beta = {
+      source  = "hashicorp/google-beta"
+      version = "3.58.0"
+    }
+  }
+}
+
+provider "google" {
+  project     = var.project_id
+  region      = var.region
+  zone        = var.zone
+  credentials = file(var.credentials_file)
+}
+
+provider "google-beta" {
+  project     = var.project_id
+  region      = var.region
+  zone        = var.zone
+  credentials = file(var.credentials_file)
+}
+
+locals {
+  vm_names                   = concat(local.admin_vm_names, local.controlplane_vm_names, local.worker_vm_names)
+  admin_vm_names             = [for vmName in var.instance_names.admin : join("-", [var.hostname_prefix, vmName])]
+  controlplane_vm_names      = [for vmName in var.instance_names.controlplane : join("-", [var.hostname_prefix, vmName])]
+  worker_vm_names            = [for vmName in var.instance_names.worker : join("-", [var.hostname_prefix, vmName])]
+  controlplane_vxlan_ips     = [for name in local.controlplane_vm_names : local.vm_vxlan_ip[name]]
+  worker_vxlan_ips           = [for name in local.worker_vm_names : local.vm_vxlan_ip[name]]
+  admin_vm_hostnames         = [for vm in module.admin_vm_hosts.vm_info : vm.hostname]
+  vm_vxlan_ip                = { for idx, vmName in local.vm_names : vmName => "10.200.0.${idx + 2}" }
+  vmHostnameToVmName         = { for vmName in local.vm_names : "${vmName}-001" => vmName }
+  cluster_yaml_file          = "${path.module}/resources/.${var.abm_cluster_id}.yaml"
+  cluster_yaml_template_file = "${path.module}/resources/anthos_gce_cluster.tpl"
+  env_var_script_template    = "${path.module}/resources/env.tpl"
+  env_var_script             = "${path.module}/resources/.env.sh"
```

Comment:
env.sh script is missing.

Reply:
Needs env script

Reply:
Ohh this was from a previous iteration I was working on. It's not required
Removed it now in 59092234b18d5c00def07e362a46ce498137ea29